### PR TITLE
Automatically find domain for the polynomial coefficients of the differential equation and support for singular initial conditions in `integrate`, `mul` and `add`.

### PR DESCRIPTION
**Pull request by shubhamtibra**

### CODE REVIEW

File: sympy/holonomic/holonomic.py [hunk 1]
```diff
@@ -2109,12 +2109,26 @@ def expr_to_holonomic(func, x=None, initcond=True, x0=0, lenics=None, domain=QQ,
...
+
+    extra_syms = [i for i in syms]
+
+    if domain == None:
+        if func.has(Float):
+            domain = RR
+        else:
+            domain = QQ
+        if len(extra_syms) != 0:
+            domain = domain[extra_syms]
```
Comment (jksuom) [inline sympy/holonomic/holonomic.py]: This should probably be made a field (using `get_field()`). The extra symbols may also occur in denominators.

Comment (shubhamtibra) [inline sympy/holonomic/holonomic.py]: I have corrected it.

File: sympy/holonomic/holonomic.py [hunk 2]
```diff
@@ -619,8 +620,39 @@ def integrate(self, limits, initcond=False):
         # to get the annihilator, just multiply by Dx from right
         D = self.annihilator.parent.derivative_operator
 
+        # if the function have initial conditions of the series format
+        if self.singular_ics:
+
```
Comment (jksuom) [inline sympy/holonomic/holonomic.py]: PEP 8 says:

     Use blank lines in functions, sparingly, to indicate logical sections.

There is quite a number in this function :)

**Comment by shubhamtibra** [comment]
This is unusual. Tests were failing locally after commit 38ccb545953318ef351cdf60b0ef65d5e4332211 and that's why I fixed some typos in the next commit. But the tests passed on travis https://travis-ci.org/sympy/sympy/builds/147237055.

**Comment by jksuom** [comment]
How would you describe the contents of this PR?

**Comment by jksuom** [comment]
It seems to me that this would solve the remaining part of #11323 .

**Comment by shubhamtibra** [comment]
Ping @jksuom @certik .

**Comment by certik** [comment]
It looks good.

**Pull request merged**